README.md

```diff
@@ -45,6 +47,23 @@ $ xargo build --target x86_64
 
 [Xargo]: https://crates.io/crates/xargo
 
+### Chapter 4
+
+We are now in long mode :tada:. Notable changes:
+
+- We were always emitting 64-bit instructions even though we are only supposed to use 32-bit
+  instructions because we start in restricted mode :scream_cat:. This has been fixed by changing the
+  target specification from a 64-bit target (`x86_64.json`) to a 32-bit one (`x86.json`).
+- All the data layout related stuff that intermezzos does in assembly has been implemented in the
+  linker script to reduce the amount of assembly. Notably, the layout of the page tables and the GDT
+  are done in the linker script.
+- Linking the page tables is done in pure Rust :+1;.
+- Actually entering long mode requires manipulation of (control) registers and those can't be
+  accessed via pure Rust, AFAIK. For that reason, this part is written in inline assembly.
+- Caveat: Because we are telling `rustc` to use a 32-bit target, `rustc` always emit 32-bit
+  instructions and we can't actually use 64-bit instructions/registers in the section of the program
+  where the CPU is already in long mode. Yikes, I'll have to think about how to solve this.
```

Comment:
I think this can be solved by splitting this crate in two: one crates that does the booting process and is compiled for a 32-bit target and the other crate that exposes the `long_mode_start` symbol and is compiled for a 64-bit target. The issue is that I think the compiler won't link together crates compiled for different types (imagine mixing a crate compiled for ARM with a crate compiled for x86). If that's the case we could compiled the `boot` crate as a staticlib and link that staticlib into the `long` crate. That should bypass the "same target" check.

All this is speculation from my part. I don't know if it will actually work.